packages/generators/templates/shared/package.json.hbs

```diff
@@ -0,0 +1,57 @@
+{
+	"name": "@automattic/{{ kebabCase name }}",
+	"version": "1.0.0",
+	"description": "{{ description }}",
+	"homepage": "https://github.com/Automattic/wp-calypso",
+	"license": "GPL-2.0-or-later",
+	"author": "Automattic Inc.",
+	"main": "dist/cjs/index.js",
+	"module": "dist/esm/index.js",
+	"calypso:src": "src/index.tsx",
+	"sideEffects": [
+		"*.css",
+		"*.scss"
+	],
+	"repository": {
+		"type": "git",
+		"url": "git+https://github.com/Automattic/wp-calypso.git",
+		"directory": "packages/{{ kebabCase name }}"
+	},
+	"publishConfig": {
+		"access": "public"
+	},
+	"bugs": "https://github.com/Automattic/wp-calypso/issues",
+	"types": "dist/types",
+	"scripts": {
+		"clean": "tsc --build ./tsconfig.json ./tsconfig-cjs.json --clean && rm -rf dist",
+		"build": "tsc --build ./tsconfig.json ./tsconfig-cjs.json && copy-assets",
+		"prepack": "yarn run clean && yarn run build",
+		"watch": "tsc --build ./tsconfig.json --watch"
+	},
+	"dependencies": {
+		"@automattic/i18n-utils": "workspace:^",
+		"classnames": "^2.3.1",
+		"react-query": "^3.32.1",
+		"tslib": "^2.3.0",
+		"utility-types": "^3.10.0",
+		"wpcom-proxy-request": "workspace:^"
+	},
+	"devDependencies": {
+		"@automattic/calypso-build": "workspace:^",
+		"@automattic/calypso-typescript-config": "workspace:^",
+		"react": "^17.0.2",
+		"react-dom": "^17.0.2",
+		"typescript": "^4.7.4"
+	},
+	"peerDependencies": {
+		"@wordpress/data": "^6.7.0",
+		"@wordpress/element": "^4.5.0",
+		"@wordpress/i18n": "^4.7.0",
+		"debug": "^4.3.3",
+		"react": "^17.0.2",
+		"react-dom": "^17.0.2",
+		"reakit-utils": "^0.15.1",
+		"redux": "^4.1.2"
```

Comment:
I think we might want to modify these somewhat. While I understand the idea, I think we should just add the bare minimum for the package to work out of the box. I don't think it's likely for every new package to need every one of these dependencies.

Additionally, some of these kind of conflict. For example, you don't need react as both a peer dependency and a dev dependency, iirc. And react should probably be a proper dependency rather than a dev dependency.

What I'd suggest is just the minimum packages needed for the build to work, and then authors can add anything else they might need

Reply:
Sounds good! I've modified the deps and created different `package.json` files for the library and component. Now I only include the react, react-dom deps to the component package. Does it make sense?